src/nodes.coffee

```diff
@@ -102,6 +136,54 @@ exports.Base = class Base
         parts.push    @makeCode ")"
     parts
 
+  compileCommentFragments: (o, node, fragments) ->
+    return fragments unless node.comments
+    # This is where comments, that are attached to nodes as a `comments`
+    # property, become `CodeFragment`s. “Inline block comments,” e.g.
+    # `/* */`-delimited comments that are interspersed within code on a line,
+    # are added to the current `fragments` stream. All other fragments are
+    # attached as properties to the nearest preceding or following fragment,
+    # to remain stowaways until they get properly output in `compileComments`
+    # later on.
+    unshiftCommentFragment = (commentFragment) ->
+      if commentFragment.unshift
+        # Find the first non-comment fragment and insert `commentFragment`
+        # before it.
+        unshiftAfterComments fragments, commentFragment
+      else
+        if fragments.length isnt 0
+          precedingFragment = fragments[fragments.length - 1]
+          if commentFragment.newLine and precedingFragment.code isnt '' and
+             not /\n\s*$/.test precedingFragment.code
+            commentFragment.code = "\n#{commentFragment.code}"
+        fragments.push commentFragment
+
+    for comment in node.comments when comment not in @compiledComments
+      @compiledComments.push comment # Don’t output this comment twice.
```

Comment:
It’s necessary because `compileWithoutComments` is only called in a handful of cases.

There are lots and lots of nodes that end up with `comments` properties that are duplicative. Like an `Assign` node where it and its children, `variable` and `value`, all have `comments` properties with the same comments. I’m not sure why this happens; it could be that all those nodes have the same location data, so the comments get attached to all of them; or when tokens go through the parser, things bubble up (like `Identifier` inside `Expression` inside `Assign`) the comments get duplicated with each pass. If there were a clean way to prevent duplication in the first place I’d prefer that, but I couldn’t figure out how to achieve that. Even if we could cut down (or eliminate) duplicated comments, this two-line check would still be a good failsafe to ensure we didn’t miss anything.